### PR TITLE
Add and improve the e2e use case of the NIC where the default route is located

### PR DESCRIPTION
**Pull request by ty-dc**
**What this PR does / why we need it**:

- Add and improve the e2e use case of the NIC where the default route is located. Assist in verifying the repair status of the PR #2411 

The changes are as follows:

1. Add a new case. Without setting any annotations or PodDefaultRouteNIC (default situation), the default route should be on net1.

2. Change the existing use case, specify PodDefaultRouteNIC as eth0, and check that the default route should be on eth0

**Which issue(s) this PR fixes**:
```
NONE
```

**make sure your commit is signed off**
Signed-off-by: ty-dc [tao.yang@daocloud.io](mailto:tao.yang@daocloud.io)

### CODE REVIEW

**Comment by ty-dc** [comment]
debug log:
```
  exec dep-name-c177c54e8b-5ff8f49c55-q48hx -n ns-6414376efd -- ip r get '8.8.8.8' 
  Execute command result:  8.8.8.8 via 10.220.144.1 dev net1  src 10.220.144.2 

  [FAILED] Expected NIC eth0 mismatch
  Expected
      <string>: 8.8.8.8 via 10.220.144.1 dev net1  src 10.220.144.2 
      
  to contain substring
      <string>: eth0
  In [It] at: /root/code/spiderpool/test/e2e/coordinator/macvlan-overlay-one/macvlan_overlay_one_test.go:302 
```

File: test/e2e/coordinator/macvlan-overlay-one/macvlan_overlay_one_test.go [hunk 1]
```diff
@@ -287,25 +288,69 @@ var _ = Describe("MacvlanOverlayOne", Label("overlay", "one-nic", "coordinator")
...
+					Expect(string(executeCommandResult)).Should(ContainSubstring(common.NIC1), "Expected NIC %v mismatch", common.NIC1)
+
+					// ip r get <IP in eth0 subnet>, should flow out from eth0
+					runGetIPString = fmt.Sprintf("ip r get %v ", ip.NextIP(net.ParseIP(pod.Status.PodIP)).String())
+					executeCommandResult, err = frame.ExecCommandInPod(pod.Name, pod.Namespace, runGetIPString, ctx)
+					GinkgoWriter.Println("Execute command result: ", string(executeCommandResult))
+					Expect(err).NotTo(HaveOccurred(), "failed to execute command, error is: %v ", err)
+					Expect(string(executeCommandResult)).Should(ContainSubstring(common.NIC1), "Expected NIC %v mismatch", common.NIC1)
+
+					// ip r get <IP in net1 subnet>, should flow out from net1
```
Comment (cyclinder) [inline test/e2e/coordinator/macvlan-overlay-one/macvlan_overlay_one_test.go:311]: 是否还可以加个 `ip r get <service_subnet>`

Comment (ty-dc) [reply test/e2e/coordinator/macvlan-overlay-one/macvlan_overlay_one_test.go:312]: done

**Comment by weizhoublue** [comment]
is this ready for merge ？

**Comment by weizhoublue** [comment]
it could  add [commit](https://github.com/spidernet-io/spiderpool/pull/2411/files) together to pr for CI

File: cmd/coordinator/cmd/utils.go [hunk 2]
```diff
@@ -429,7 +429,11 @@ func (c *coordinator) tunePodRoutes(logger *zap.Logger, configDefaultRouteNIC st
 
 		if configDefaultRouteNIC == c.currentInterface {
 			for idx, route := range defaultInterfaceRoutes {
-				if route.Dst != nil {
+				zeroIPAddress := net.IPv4zero
+				if defaultInterfaceRoutes[idx].Family == netlink.FAMILY_V6 {
+					zeroIPAddress = net.IPv6zero
+				}
+				if route.Dst != nil || !route.Dst.IP.Equal(zeroIPAddress) {
```
Comment (cyclinder) [inline cmd/coordinator/cmd/utils.go:436]:
```
if !route.Dst.IP.Equal(zeroIPAddress) {
```

File: cmd/coordinator/cmd/utils.go [hunk 3]
```diff
@@ -453,7 +457,11 @@ func (c *coordinator) tunePodRoutes(logger *zap.Logger, configDefaultRouteNIC st
 
 		} else if configDefaultRouteNIC == podDefaultRouteNIC {
 			for idx, route := range currentInterfaceRoutes {
-				if route.Dst != nil {
+				zeroIPAddress := net.IPv4zero
+				if defaultInterfaceRoutes[idx].Family == netlink.FAMILY_V6 {
+					zeroIPAddress = net.IPv6zero
+				}
+				if route.Dst != nil || !route.Dst.IP.Equal(zeroIPAddress) {
```
Comment (cyclinder) [inline cmd/coordinator/cmd/utils.go:464]:
```
if !route.Dst.IP.Equal(zeroIPAddress) {
```

**Review by cyclinder — CHANGES_REQUESTED** [review]

File: cmd/coordinator/cmd/utils.go (hunk 2 above)
Comment (ty-dc) [reply cmd/coordinator/cmd/utils.go]: done

File: cmd/coordinator/cmd/utils.go (hunk 3 above)
Comment (ty-dc) [reply cmd/coordinator/cmd/utils.go]: done

File: test/Makefile [hunk 4]
```diff
@@ -179,6 +179,15 @@ setup_spiderpool:
 		fi ; \
 		if [ "$(E2E_SPIDERPOOL_ENABLE_COORDINATOR)" == "true" ] ; then \
 		    HELM_OPTION+=" --set coordinator.enabled=true " ; \
+			if [ "$(INSTALL_CILIUM)" == "true" ] ; then \
```
Comment (cyclinder) [inline test/Makefile:182]: Actually, This is the same problem as https://github.com/spidernet-io/spiderpool/issues/2433. It was fixed by https://github.com/spidernet-io/spiderpool/pull/2434.  it's OK to merge this part of the code for me. But we should revert these changes until https://github.com/spidernet-io/spiderpool/pull/2434 has been merged.

**Review by cyclinder — APPROVED** [review]

File: test/Makefile (hunk 4 above)
Comment (ty-dc) [reply test/Makefile:182]: ok
